python/neuroglancer/downsample.py

```diff
@@ -23,9 +23,21 @@ def method(layer_type):
     return downsample_with_averaging
   elif layer_type == 'segmentation':
     return downsample_segmentation
+  elif layer_type == 'activation':
+    return downsample_with_max_pooling
   else:
     return downsample_with_striding 
 
+def validate_factor(array, factor):
+  factor = np.array(factor, dtype=np.int32)
+  if np.any(factor <= 0):
+    raise ValueError("Factors less than one don't make sense. Factor: {}".format(factor))
+
+  if len(array.shape) == 4 and len(factor) == 3:
```

Comment:
My suggestions here are either:
1.  rule based magic: no magic numbers. if `len(factor) < len(array.shape)` then append 1's to factor to avoid downsampling other dimensions
or
2. no magic: throw an error if `len(factor) != len(array.shape)`. force the user to enter the correct value manually to ensure they get exactly what they ask for

My preference is for 2 here